README.md

```diff
@@ -154,6 +154,16 @@ sbt-ghpages will only delete files which are matched by the FileFilter specified
 For example, to prevent sbt-ghpages from deleting the "CNAME" file located at the root of your site, and any file
 named "versions.html", add the following to your build.sbt:
 
+## Maintaining Old Versions
+By setting just `ghpagesKeepVersions` flag, you can maintain history of all your old and current versions of site/documents.
+Behind the scenes, this flag plays two important roles:
+1. Set's the `includeFilter in ghpagesCleanSite` to include all the files from current versions. 
+    (That means, if you publish same version again, old content from that version will be replaced by new content)
+2. While publishing site to gh-pages, new directory with current version's name is created and all the site contents will be
```

Comment:
I think the fact that all the mappings will **all** be prefixed with the version number should be more emphasized.

Also, how would you recommend to do to have a landing page that automatically points to the latest version?

Reply:
For automatic pointing, we can copy latest version at top level as well as inside respective version directory. This can be filtered for snapshot versions which can reside only inside respective snapshot dirs.
We are doing exactly same here:
https://github.com/tmtsoftware/sbt-docs/blob/master/src/main/scala/org/tmt/sbt/docs/Settings.scala

Reply:
> we can copy

You mean sbt-doc or the users of sbt-doc?

Reply:
In the `GhpagesPlugin`, when creating `ghpagesPrivateMapping` check can be added to detect if its a SNAPSHOT version or not and based on that generate mappings.  Something like below:
 
```scala

  private def ghpagesPrivateMappingsTask = Def.task {
    val isSnapshot = version.value.contains("SNAPSHOT")
    val makeSiteMappings = (mappings in SitePlugin.autoImport.makeSite).value
    
    val updatedMappings =
      if (ghpagesKeepVersions.value) {
        val updated = makeSiteMappings map { case (file, target) => (file, 
        version.value + "/" + target) }
        if(isSnapshot) upadated
        else makeSiteMappings ++ updated
      }
      else makeSiteMappings

    updatedMappings
  }
```

Reply:
`sbt-doc` uses `ghpages` plugin, I have added workaround I mentioned above in `sbt-doc` plugin. As currently `ghpages` plugin does not support versioning.

Reply:
OK. Do we want to add a setting key letting users control whether they want to have their website at the root in addition to have it prefixed by the version number?

Reply:
Sure, make sense!

Reply:
I have incorporated all these points in my last commit, let me know if those make sense!